cli/util/util.go

```diff
@@ -26,3 +31,39 @@ func Dial(addr, certPEM string) (*grpc.ClientConn, error) {
 		grpc.WithDefaultCallOptions(grpc.UseCompressor(gzip.Name)),
 		grpc.WithUnaryInterceptor(errors.UnaryClientInterceptor))
 }
+
+func GetDockerClient() (*client.Client, error) {
+	dockerClient, dcErr := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+
+	// If a custom host is set, don't try any funny business.
+	if dcErr != nil || dockerClient.DaemonHost() != client.DefaultDockerHost {
+		return dockerClient, dcErr
+	}
+
+	socketUrl, err := url.Parse(dockerClient.DaemonHost())
+	if err != nil {
+		// We really do not expect this to ever happen.
+		panic(err)
+	}
+
+	if socketUrl.Scheme != "unix" {
+		return dockerClient, dcErr
```

Comment:
`dcErr` is always `nil` in these `return`s right? I think it'd be easier to reason about if you just return `nil`.